front/lib/api/workspace.ts

```diff
@@ -117,3 +153,300 @@ export async function getPendingInvitations(
     };
   });
 }
+
+/**
+ * Deletes all workspace related data from the db
+ * Requires the authenticator to be a super admin.
+ */
+export async function dangerousSuperAdminDeleteWorkspaceData({
+  auth,
+}: {
+  auth: Authenticator;
+}): Promise<void> {
+  // Verify auth is super admin and workspace exists.
+  if (!auth.isDustSuperUser()) {
+    throw new Error("Unauthorized.");
+  }
+  const workspace = auth.workspace();
+  if (!workspace) {
+    throw new Error("Workspace does not exist.");
+  }
+
+  // Workspace must have no managed data sources.
+  const dataSources = await getDataSources(auth);
+  if (dataSources.length > 0) {
+    throw new Error("Workspace has data sources. Delete them first.");
+  }
+
+  // For now we don't support deleting workspaces who had a paid subscription at some point.
+  const subscriptions = await Subscription.findAll({
+    where: {
+      workspaceId: workspace.id,
+      stripeSubscriptionId: {
+        [Op.not]: null,
+      },
+    },
+  });
+  if (subscriptions.length > 0) {
+    throw new Error(
+      "Workspace has/had a paid subscription. We do not support deleting such workspaces."
+    );
+  }
+
+  // WE DO NOT RELY ON CASCADE DELETION.
+  await front_sequelize.transaction(async (t) => {
+    await _deleteConversations(workspace, t);
+    await _deleteAgents(workspace, t);
+    await Promise.all([
+      _deleteApps(workspace, t),
+      _deleteInvitations(workspace, t),
+      _deleteMembershipsAndUserIfOneOnly(workspace, t),
+    ]);
+    await _deleteWorkspace(workspace, t);
+  });
+  return;
+}
+
+const _deleteConversations = async (
+  workspace: WorkspaceType,
+  t: Transaction
+) => {
+  const conversations = await Conversation.findAll({
```

Comment:
We likely want to chunk it and parallelize each chunk? This would take forever otherwise on a large workspace no?

Reply:
yep